subiquity/server/controllers/ad.py

```diff
@@ -34,6 +39,92 @@ async def GET(self) -> Optional[ADConnectionInfo]:
         return self.model.conn_info
 
     async def POST(self, data: ADConnectionInfo) -> List[ADValidationResult]:
+        """ Configures this controller with the supplied info.
```

Comment:
Perhaps I should have made this comment on the spec or on the last PR or something but well.

Doing this validation as part of POST seems odd. The pattern in the TUI (and just general UX principle) is that invalid data is highlighted in the client as soon as it is entered (or at least as soon as the relevant input widget loses focus) and the "submit" button is not available until the data passes validation.

I can understand the desire to do validation in the server so that it doesn't have to be duplicated between client and server and indeed checking that the data submitted in POST is valid, but the API here does not seem sufficient to provide a good experience. Am I missing something?

Reply:
I think you raised a good point. I should have considered it in the spec as well. I'll create specific endpoints clients call immediately after the user finishes input a field, allowing better UX. I was only considering doing so for the DC name, because, besides the regex validation, there will be an attempt to PING it at next. But it makes sense to create the endpoints right away.

Reply:
I like that we have client-side validation but I think we should still redo the validation on the server-side, especially because we're seemingly trying to avoid directory traversals issues.

Reply:
The web design best practice is that you do client-side validation for a smooth experience, and server-side validation because trusting the client to only send valid data is how you end up with sql injections and other similar nonsense.

We don't have quite the same level of urgency in Subiquity, as the client is trusted, but validating in both places is the safest choice.